src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4983,7 +5002,26 @@ public override bool PreVisit(CallNode node)
                 _currentScope = scopeNew.Parent;
                 PostVisit(node.Args);
 
-                // Typecheck the invocation.
+                if (maybeFunc.HasIdentifiers && _features.HasFlag(Features.SupportIdentifiers))
+                {
+                    var i = 0;
+
+                    foreach (var arg in args)
+                    {
+                        i++;
```

Comment:
if we're using a counter anyway, should we just do 
`for (i = 0; i < args.Length; i++)` instead of allocating an iterator?

Reply:
Not convinced that calling args[i] is more efficient.
Anyway for a loop with 3 or 4 args at max, will it make any difference?

Reply:
Fair enough, but if you're using a counter anyway, it also looks cleaner, imo.